kip-0023.md

```diff
@@ -0,0 +1,425 @@
+---
+KIP: "0023"
+Title: WebAuthn signatures for Chainweb and Pact
+Author: Andy Tang @EnoF, Eileen Guo @eileenmguo
+Status: Draft
+Type: Standard
+Category: Interface
+Created: 2023-10-27
+---
+
+# Abstract
+
+This document describes the proposal to leverage WebAuthn signatures as an
+alternative to ED25519 signatures, paving the way for users to create
+passwordless accounts and approve transactions without managing public and
+private keys. With this proposal, transactions from Pact smart contracts that
+are processed by Chainweb nodes can be signed with biometric authentication.
+
+# Motivation
+
+The current user experience for creating an account and signing transactions can
+be challenging for users unfamiliar with blockchain technology:
+
+- The concept of public and private key pairs is not widely understood.
+- Wallets often generate a 12 to 24 word mnemonic phrase that users need to
+  store securely in addition to a traditional password.
+
+People often underestimate the importance of keeping their private keys secure,
+and are likely to lose, mistype, or unintentionally expose them. Wallets also
+typically access private keys through software that, if compromised, might be
+used to sign transactions without a user's consent. With Kadena SpireKey and
+WebAuthn signatures, key pairs are stored directly on a secure enclave on the
+user's own devices. Storing keys in this way prevents the private key from being
+exposed or lost. In addition, Kadena SpireKey prompts users to sign for every
+transaction, providing greater security, transparency, and control.
+
+Similarly, mnemonic phrases are difficult to store securely and easy to lose.
+With support for WebAuthn signatures, Kadena SpireKey enables users to securely
+generate and store key pairs directly on their own hardware devices, without
+needing to remember or store a mnemonic phrase. The key pairs stored on devices
+can typically be accessed using biometric authentication—like Touch ID or Face
+ID—allowing users to log in and sign transactions without passwords.
+Additionally, WebAuthn has a convienient service provided to sync the key pairs
+to the cloud, in the form of `Passkeys`. This service allows users to make use
+of any `Passkey Provider` of their choice, removing the reliance on big
+corporations to store key pairs.
+
+# Specifications
+
+This section provides technical specifications for the implementation of this
+proposal.
+
+## Account creation
+
+The `coin` contract removed the ability to rotate keysets since `v6`. This
```

Comment:
```suggestion
The `coin` contract removed the ability to rotate the governance predicate (e.g. a keyset) associated with a principal account since `v6`. This
```